src/imitation/rewards/discrim_net.py

```diff
@@ -277,159 +199,149 @@ def __init__(self, reward_net: reward_net.RewardNet, entropy_weight: float = 1.0
           scale: should inputs be rescaled according to declared observation
             space bounds?
         """
-        # for serialisation
-        args = dict(locals())
-
-        # things we'll need in .build_graph()
-        self._observation_space = observation_space
-        self._action_space = action_space
-        self._scale = scale
-        self._build_discrim_net = build_discrim_net
-        self._build_discrim_net_kwargs = build_discrim_net_kwargs or {}
-
-        self._disc_mlp = None
-        # Builds graph via call to `self.build_graph()`.
-        DiscrimNet.__init__(self)
-        assert self._disc_mlp is not None
+        super().__init__(
+            observation_space=observation_space, action_space=action_space, scale=scale
+        )
 
-        # records args for un-pickling as well as newly-created model
-        serialize.LayersSerializable.__init__(**args, layers=self._disc_mlp)
+        if discrim_net is None:
+            self.discriminator = ActObsMLP(
+                action_space=action_space,
+                observation_space=observation_space,
+                hid_sizes=(32, 32),
+            )
+        else:
+            self.discriminator = discrim_net
 
         logging.info("using GAIL")
 
-    @property
-    def obs_ph(self):
-        return self._obs_ph
-
-    @property
-    def act_ph(self):
-        return self._act_ph
-
-    @property
-    def next_obs_ph(self):
-        return self._next_obs_ph
-
-    @property
-    def done_ph(self):
-        return self._done_ph
+    def logits_gen_is_high(
+        self,
+        state: th.Tensor,
+        action: th.Tensor,
+        next_state: th.Tensor,
+        done: th.Tensor,
+        log_policy_act_prob: Optional[th.Tensor] = None,
+    ) -> th.Tensor:
+        """Compute the discriminator's logits for each state-action sample.
 
-    def build_graph(self):
-        phs, inps = networks.build_inputs(
-            self._observation_space, self._action_space, scale=self._scale
-        )
-        self._obs_ph, self._act_ph, self._next_obs_ph, self._done_ph = phs
-        self.obs_input, self.act_input, _, self.done_input = inps
+        A high value corresponds to predicting generator, and a low value corresponds to
+        predicting expert.
+        """
+        logits = self.discriminator(state, action)
+        return logits
 
-        with tf.variable_scope("discrim_network"):
-            self._disc_logits_gen_is_high, self._disc_mlp = self._build_discrim_net(
-                [self.obs_input, self.act_input], **self._build_discrim_net_kwargs
-            )
-        self._policy_test_reward = self._policy_train_reward = -tf.log_sigmoid(
-            self._disc_logits_gen_is_high
-        )
+    def reward_test(
+        self,
+        state: th.Tensor,
+        action: th.Tensor,
+        next_state: th.Tensor,
+        done: th.Tensor,
+    ) -> th.Tensor:
+        rew = self.reward_train(state, action, next_state, done)
```

Comment:
It's here in case somebody overrides `RewardNet` and forgets to squeeze the output (like I did the first time I wrote `RewardNet`…). Maybe paranoid, but "extra unit dimension" bugs are exceptionally hard to track down after the fact.